onnx/defs/tensor/defs.cc

```diff
@@ -18,7 +19,7 @@ NOTE: Casting to and from strings is not supported yet.
         "to",
         "The data type to which the elements of the input tensor are cast."
         "Strictly must be one of the types from DataType enum in TensorProto",
-        AttributeProto::STRING)
+        AttributeProto::INT)
```

Comment:
as we discussed in person, I believe that the correct way to do this is to add a case to the attribute type enum, which corresponds to TensorProto::DataType, rather than using an integer that happens to line up with the integer values of the DataType enum.

The advantages are that (1) it's clearer, and (2) tooling which understands protobuf will automatically work - for example the `str()` of a `ModelProto` in python will automatically say `TensorProto.FLOAT` instead of `4`.

But if i'm the lone voice here then I won't hold it up

Reply:
Having the visualization tooling support is indeed more ideal, however the cost is not small:
1. each backend needs to be updated to acknowledge a new AttributeProto data type
2. existing tools (checker, our c++ IR, schema declarations) needs to be updated to acknowledge this as well

And use cases seem to be small, so far I can only think of Cast and GivenTensorFill (experimental).